lib/publiccloud/instance.pm

```diff
@@ -0,0 +1,43 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Base class for public cloud instances
+#
+# Maintainer: Clemens Famulla-Conrad <cfamullaconrad@suse.de>
+
+package publiccloud::instance;
+use testapi;
+use Mojo::Base -base;
+
+has instance_id => undef;               # unique CSP instance id
+has ip          => undef;               # public IP of instance
```

Comment:
can we add 2 variables here for network?

public_ip
private_ip

In case we need to run tests with multimachine

Reply:
yes we can extend it as much as needed. But maybe we should do it, when we actually fill it, or?